dd-sdk-android-core/src/test/kotlin/com/datadog/android/utils/forge/DNSErrorStatusForgeryFactory.kt

```diff
@@ -0,0 +1,18 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.utils.forge
+
+import com.datadog.android.core.internal.data.upload.UploadStatus
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.ForgeryFactory
+
+internal class DNSErrorStatusForgeryFactory : ForgeryFactory<UploadStatus.DNSError> {
+
+    override fun getForgery(forge: Forge): UploadStatus.DNSError {
+        return UploadStatus.DNSError
```

Comment:
why we need a factory here which only returns one single type of enum ?

Reply:
It's not an enum but an object as part of a `sealed class`

Reply:
even so in this case this factory only returns one value of that `UploadStatus` sealed class. Wouldn't it be more simple to just add this value in your test data without using forge ?